src/main/java/server/Server.java

```diff
@@ -114,47 +115,18 @@ public void run() {
 
         @Override
         public void run() {
-            ExecutorService executor = Executors.newSingleThreadExecutor();
-
+            System.out.println("Started reading");
             while (true) {
-                Future<ClientToServer> futureMsg = executor.submit(new Callable() {
-                    @Override
-                    public ClientToServer call() throws Exception {
-                        try {
-                            return ClientToServer.parseDelimitedFrom(Server.this.client.getInputStream());
-                        }
-                        catch (IOException e) {
-                            System.out.println("IO Exception: " + e);
-                            throw new RuntimeException("Failed getting input stream");
-                        }
-                    }
-                });
-
                 try {
-                    Server.this.msgFromClient = futureMsg.get(3000, TimeUnit.MILLISECONDS);
-
-                    if (Server.this.msgFromClient == null) {
-                        throw new NullPointerException();
```

Comment:
Watch out, throwing this `NullPointerException` here was how the old base-station detected if the pod disconnected. Running the current version of the base-station doesn't detect when the pod has disconnected, try seeing if there's a way to tell if `msgFromClient` is empty or something similar!